include/tvm/runtime/memory.h

```diff
@@ -124,11 +145,69 @@ class SimpleObjAllocator :
   };
 };
 
+// Array allocator that uses new/delete.
+class ArrayObjAllocator :
+      public ObjAllocatorBase<ArrayObjAllocator> {
+ public:
+  template<typename ArrayType, typename ElemType>
+  class Handler {
```

Comment:
Let us move rename it as  ArrayHandler and move it into SimpleObjAllocator. 

Rationale: In the future, we want to be able to create different Allocators, and call make_inplace_array_object in these allocators. It is easier to have a single allocator object for both array object and normal objects

Reply:
makes sense.